language/stdlib/modules/0L/FIFO.move

```diff
@@ -0,0 +1,76 @@
+address 0x1 {
+
+//Implementation of a FIFO that utilizes two vectors
+//Achieves Amortized O(1) cost per operation 
+//CAUTION: In worst case this can result in O(n) cost, adjust gas allowance accordingly 
+module FIFO {
+    use 0x1::Vector;
+
+
+    struct FIFO<Element> {
+        incoming: vector<Element>,
+        outgoing: vector<Element>
+    }
+    public fun empty<Element>(): FIFO<Element>{
+        let incoming = Vector::empty<Element>();
+        let outgoing = Vector::empty<Element>();
+        FIFO {
+            incoming:incoming,
+            outgoing:outgoing,
+        }
+    }
+
+    public fun push<Element>(v: &mut FIFO<Element>, new_item: Element){
+        Vector::push_back<Element>(&mut v.incoming, new_item);
+    }
+
+    public fun push_LIFO<Element>(v: &mut FIFO<Element>, new_item: Element){
```

Comment:
Maybe I missed it but I didn't see a use of push_LIFO in autopay.

Reply:
It's used in LibraAccount when an escrowed payment is larger than the account limit. It gets popped off the queue, some of the payment is sent, then the remainder is pushed back onto the queue, but at the front (LIFO) with this function.

Reply:
good